src/digraph-cyclicity.jl

```diff
@@ -0,0 +1,371 @@
+abstract Visitor
+
+"""
+```transitiveclosure!(dg::DiGraph)```
+
+Compute the transitive closure of a directed graph, using the Floyd-Warshall
+algorithm.
+"""
+function transitiveclosure!(dg::DiGraph)
+    for k in dg.vertices
+        for i in dg.vertices
+            for j in dg.vertices
+                if (has_edge(dg, i, k) & has_edge(dg, k, j))
```

Comment:
I understand - thanks for your patience and for the detailed explanation. It's the same as floyd-warshall in terms of efficiency, so the only optimization would be to use Dijkstra on sparse graphs.

OK. Let's change the internals to use the accessors (vertices(dg) vs dg.vertices), and put a note in the comments that this is O(V^3) and we're good to go.